src/net/mormot.net.ldap.pas

```diff
@@ -1179,6 +1241,42 @@ procedure TLdapAttributeList.Delete(const AttributeName: RawUtf8);
 
 { TLdapResult }
 
+function TLdapResult.CopyObjectSid(out objectSid: TSid): Boolean;
+var
+  SidAttr: TLdapAttribute;
+  SidBinary: RawByteString;
+begin
+  SidAttr := Attributes.Find('objectSid');
+  if not Assigned(SidAttr) then
+    Exit(False);
+
+  SidBinary := SidAttr.GetRaw;
+  Move(SidBinary[1], objectSid, Length(SidBinary));
+  Result := True;
+end;
+
+function TLdapResult.GetObjectGUID: PGuid;
+var
+  GUIDAttr: TLdapAttribute;
+begin
+  GUIDAttr := Attributes.Find('objectGUID');
+  if Assigned(GUIDAttr) then
+    Result := @(RawByteString(GUIDAttr.GetRaw)[1])

```

Comment:
it will GPF randomly
GetRaw returns a local RawByteString variable, which is returned as a PGuid
but this local RawByteString is released as soon as the function leaves
so the returned pointer will be invalid sooner or later